dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/TrackerErrorCode.java

```diff
@@ -0,0 +1,153 @@
+        "An Enrollment cannot be created into Program without registration." ),
+    E1020( "Enrollment date: `{0}`, can`t be future date." ),
+    E1021( "Incident date: `{0}`, can`t be future date." ),
+    E1023( "DisplayIncidentDate is true but IncidentDate is null or invalid formatted: `{0}`." ),
+    E1025( "Invalid Enrollment date: `{0}`." ),
+    E1041( "Enrollment OrganisationUnit: `{0}`, and Program: `{1}`, OrganisationUnit: `{2}`, don't match." ),
+    E1068( "Could not find TrackedEntityInstance: `{0}`, linked to Enrollment." ),
+    E1069( "Could not find Program: `{0}`, linked to Enrollment." ),
+    E1070( "Could not find OrganisationUnit: `{0}`, linked to Enrollment." ),
+    E1080( "Enrollment: `{0}`, already exists." ),
+    E1081( "Enrollment: `{0}`, do not exist." ),
+    E1030( "Event: `{0}`, already exists." ),
+    E1032( "Event: `{0}`, do not exist." ),
+    E1035( "Event: `{0}`, ProgramStage value is NULL." ),
+    E1086( "Event: `{0}`, has a program: `{1}`, that is a registration but its ProgramStage is not valid or missing." ),
+    E1088( "Event: `{0}`, program: `{1}`, and ProgramStage: `{2}`, could not be found." ),
+    E1089( "Event: `{0}`, ProgramStage Program and Event Program don't match." ),
+    E1000( "User: `{0}`, has no write access to OrganisationUnit: `{1}`." ),
+    E1001( "User: `{0}`, has no data write access to TrackedEntityType: `{1}`." ),
+    E1091( "User: `{0}`, has no data write access to Program: `{1}`." ),
+    E1095( "User: `{0}`, has no data write access to ProgramStage: `{1}`." ),
+    E1096( "User: `{0}`, has no data read access to Program: `{1}`." ),
+    E1100( "User: `{0}`, is lacking 'F_TEI_CASCADE_DELETE' authority to delete TrackedEntityInstance: `{1}`." ),
+    E1102( "User: `{0}`, does not have access to the tracked entity: `{1}`, Program: `{2}`, combination." ),
+    E1103( "User: `{0}`, is lacking 'F_ENROLLMENT_CASCADE_DELETE' authority to delete Enrollment : `{1}`." ),
+    E1104( "User: `{0}`, has no data read access to program: `{1}`, TrackedEntityType: `{2}`." ),
+    E1112( "Attribute value: `{0}`, is set to confidential but system is not properly configured to encrypt data." ),
+    E1118( "Note does not have a valid uuid: `{0}`." ),
+    E1119( "Note does not have a value: `{0}`." ),
+    E1121( "Note stored at date is invalid: `{0}`; Error: `{1}`." ),
+    E1055( "Default AttributeOptionCombo is not allowed since program has non-default CategoryCombo." ),
+    E1115( "Could not find CategoryOptionCombo: `{0}`." ),
+    E1116( "Could not find CategoryOption: `{0}`." ),
+    E1117( "CategoryOptionCombo does not exist for given category combo and category options: `{0}`." ),
+    E1099( "User: `{0}`, has no write access to CategoryOption: `{1}`." ),
+    E1039( "ProgramStage: `{0}`, is not repeatable and an event already exists." ),
+    E1048( "Object: `{0}`, uid: `{1}`, has an invalid uid format." ),
+
+    // TODO: Delete not working yet
```

Comment:
@netroms can you note down this as well for the next meeting. Delete + TODO error usage comments.

Reply:
Done